### PR TITLE
Other common 1.0 API breaks

### PR DESCRIPTION
**Pull request by kdorosh**

### CODE REVIEW

File: docs/content/upgrading/1.0.0.md [hunk 1]
```diff
@@ -39,7 +39,17 @@ https://github.com/solo-io/gloo/pull/1697
...
+upstreams. Virtual services referencing static upstreams must now manually set `auto_host_rewrite` to true to preserve
+old behavior. https://github.com/solo-io/gloo/pull/1341
+   - By default (i.e., `virtual_services` and `virtual_service_selector` are omitted on the Gateway), Gloo now only
+propagates config from Virtual Services in the same namespace as the referencing Gateway (compare in
+[0.21.1](https://github.com/solo-io/gloo/blob/v0.21.1/projects/gateway/api/v2/gateway.proto#L84-L91) to
+[1.2.0](https://github.com/solo-io/gloo/blob/696af5be30284847b37b4c9f12ecd42c1c88d433/projects/gateway/api/v1/gateway.proto#L81-L99)).
+Configuration to enable `virtual_service_selector` to reference virtual services outside the referencing Gateway's namespace was added [here](https://github.com/solo-io/gloo/pull/1763)
+in Gloo 1.2.0.
+   - Function Discovery Service (FDS) now defaults to whitelist mode rather than blacklist mode; [FDS config docs]({{< versioned_link_path fromRoot="/advanced_configuration/fds_mode/#configuring-the-fdsmode-setting" >}}).
+   - All instances of `...plugins` in our API (e.g., `virtualHostPlugins`, `routePlugins`) has been renamed to `options`.
```
Comment (GrahamGoudeau) [inline docs/content/upgrading/1.0.0.md]: this is a little ambiguous- I can see someone interpreting the end result of this change as either:

* the field is named `virtualHostOptions`
* the field is just named `options`

File: docs/content/upgrading/1.0.0.md [hunk 2]
```diff
@@ -39,7 +39,17 @@ https://github.com/solo-io/gloo/pull/1697
 CRD has had the `.v2` dropped from its name and API group; the
 `gateway-resource-reader` RBAC Role has had the corresponding rule on the `gateway.solo.io.v2` API
 group removed. https://github.com/solo-io/gloo/pull/1666
-
+   - The route config `auto_host_rewrite` is no longer implicitly set on virtual services that reference static
+upstreams. Virtual services referencing static upstreams must now manually set `auto_host_rewrite` to true to preserve
+old behavior. https://github.com/solo-io/gloo/pull/1341
+   - By default (i.e., `virtual_services` and `virtual_service_selector` are omitted on the Gateway), Gloo now only
```
Comment (GrahamGoudeau) [inline docs/content/upgrading/1.0.0.md]: the parenthetical in this line is confusingly placed, I think. Maybe at the end, something like 

```
(note that `virtual_services` and `virtual_service_selector` are omitted on Gateway resources)
```
something along those lines?

Comment (kdorosh) [reply docs/content/upgrading/1.0.0.md]: I think this reads clearer as is

**Pull request merged**